### PR TITLE
Action system

### PR DESCRIPTION
**Pull request by Fogapod**
Streamlined all major actions, they now use distinct steps

### CODE REVIEW

File: usautobuild/actions/api_caller.py [hunk 1]
```diff
@@ -39,4 +36,4 @@ def version_to_date(version: str) -> str:
         month = version[2:4]
         day = version[4:6]
 
-        return f"20{year}-{month}-{day}"
+        return f"20{year:0>2}-{month:0>2}-{day:0>2}"
```
Comment (corp-0) [inline usautobuild/actions/api_caller.py:39]: What's this

Comment (Fogapod) [reply usautobuild/actions/api_caller.py:39]: 1 will become 01

Comment (corp-0) [reply usautobuild/actions/api_caller.py:39]: It wasn't doing that already?

Comment (Fogapod) [reply usautobuild/actions/api_caller.py:39]: right, inputs are strings already, not numbers. i will undo this

**Comment by Fogapod** [comment]
I don't remember what this is

**Pull request closed without merging**